res/controllers/midi-controls-0.0.js

```diff
@@ -0,0 +1,668 @@
+            engine.setParameter(this.group, this.outCo, value);
+        },
+        outGetValue: function () {
+            return engine.getValue(this.group, this.outCo);
+        },
+        outSetValue: function (value) {
+            engine.setValue(this.group, this.outCo, value);
+        },
+        outToggle: function () {
+            this.outSetValue( ! this.outGetValue());
+        },
+
+        connect: function () {
+            /**
+            Override this method with a custom one to connect multiple Mixxx COs for a single Control.
+            Add the connection objects to the this.connections array so they all get disconnected just
+            by calling this.disconnect(). This can be helpful for multicolor LEDs that show a
+            different color depending on the state of different Mixxx COs. Refer to
+            SamplerButton.connect() and SamplerButton.output() for an example.
+            **/
+            if (undefined !== this.group &&
+                undefined !== this.outCo &&
+                undefined !== this.output &&
+                typeof this.output === 'function') {
+                this.connections[0] = engine.connectControl(this.group, this.outCo, this.output);
+            }
+        },
+        disconnect: function () {
+            if (this.connections[0] !== undefined) {
+                this.connections.forEach(function (connection) {
+                    connection.disconnect();
+                });
+            }
+        },
+        trigger: function() {
+            engine.trigger(this.group, this.outCo);
+        },
+        shiftOffset: 0,
+        sendShifted: false,
+        shiftChannel: false,
+        shiftControl: false,
+        send: function (value) {
+            if (this.midi === undefined || this.midi[0] === undefined || this.midi[1] === undefined) {
+                return;
+            }
+            midi.sendShortMsg(this.midi[0], this.midi[1], value);
+            if (this.sendShifted) {
+                if (this.shiftChannel) {
+                    midi.sendShortMsg(this.midi[0] + this.shiftOffset, this.midi[1], value);
+                } else if (this.shiftControl) {
+                    midi.sendShortMsg(this.midi[0], this.midi[1] + this.shiftOffset, value);
+                }
+            }
+        },
+    };
+
+    var Button = function (options) {
+        Control.call(this, options);
+    };
+    Button.prototype = new Control({
+        onlyOnPress: true,
+        on: 127,
+        off: 0,
+        inValueScale: function () { return ! this.inGetValue(); },
+        separateNoteOnOff: false,
+        input: function (channel, control, value, status, group) {
+            if (this.onlyOnPress) {
+                var pressed;
+                if (this.separateNoteOnOff) {
+                    // Does the first nybble of the first MIDI byte indicate a
+                    // note on or note off message?
+                    pressed = (status & 0xF0) === 0x90;
```

Comment:
I don't have any controllers that send these kinds of signals for button press/release, so I haven't tested this.